gradle/java.gradle

```diff
@@ -218,62 +190,18 @@ def isJavaVersionAllowed(JavaVersion version) {
   return true
 }
 
-/**
- * For a given java home return the location of java executable
- */
-static String toExecutable(String javaHome) {
-  return Objects.requireNonNull(javaHome) + "/bin/java"
-}
-
-/**
- * Returns java home for a given version or {@code null} if not found
- */
-String findJavaHome(JavaVersion version) {
-  def javaHome = System.getenv("JAVA_${version.majorVersion}_HOME")
-  if (javaHome == null) {
-    return null
+[8, 15].findAll { isJavaVersionAllowed(JavaVersion.toVersion(it)) }.each {version ->
+  def testJavaForVersion = tasks.register("testJava${version}", Test) {
+    javaLauncher = javaToolchains.launcherFor {
+      languageVersion = JavaLanguageVersion.of(version)
+    }
   }
-  def foundVersion = getJavaHomeVersion(javaHome)
-  return version == foundVersion ? javaHome : null
-}
-
-ext {
-  findJavaHome = this.&findJavaHome
-  toExecutable = this.&toExecutable
-}
-
-def addTestRule(String testTaskName) {
-  def prefix = testTaskName + "Java"
-  tasks.addRule("Pattern: $prefix<Version>: Runs tests using given java version") { String taskName ->
-    if (taskName.startsWith(prefix)) {
-      def requestedJavaVersion = JavaVersion.toVersion(taskName - prefix)
-      def gradleJavaVersion = JavaVersion.current()
-
-      if (gradleJavaVersion != requestedJavaVersion) {
-        def javaHomeForTests = findJavaHome(requestedJavaVersion)
-        if (javaHomeForTests != null) {
-
-          tasks.withType(Test).all {
-            executable = toExecutable(javaHomeForTests)
-            enabled = isJavaVersionAllowed(requestedJavaVersion)
-          }
-        } else {
-          throw new BuildCancelledException("Requested java version $requestedJavaVersion not found")
-        }
-      }
 
-      task(taskName) {
-        if (project.tasks.findByName(testTaskName) != null) {
-          dependsOn testTaskName
-        }
-      }
-    }
+  if (rootProject.findProperty('testAdditionalJavaVersions') == 'true') {
```

Comment:
I added this like the `-java` repo has since it doesn't hurt. Here, the tradeoff of test vs build time is so high that it probably doesn't make sense to take advantage of "build once, test many" that we do there, and instead continue to matrix in GitHub resulting in "build + test many" which is fine.

Reply:
I don't see where do we specify this property?

Reply:
Thought it could be useful from command line but on second thought guess not so much.